crates/zksync/core/src/lib.rs

```diff
@@ -214,15 +214,15 @@ pub fn try_decode_create2(data: &[u8]) -> Result<(H256, H256, Vec<u8>)> {
 }
 
 /// Compute a CREATE address according to zksync
-pub fn compute_create_address(sender: Address, nonce: u64) -> Address {
-    get_create_zksync_address(sender.to_h160(), Nonce(nonce as u32)).to_address()
+pub fn compute_create_address(sender: Address, nonce: u32) -> Address {
+    get_create_zksync_address(sender.to_h160(), Nonce(nonce)).to_address()
 }
 
 /// Compute a CREATE2 address according to zksync
 pub fn compute_create2_address(
     sender: Address,
     bytecode_hash: H256,
-    salt: B256,
+    salt: H256,
```

Comment:
just asking, why H256 here? It's my understanding this is generally regarded as a B256 value, at least from alloy

Reply:
Not a hard opinion, only because this function is defined and called within zksync context which uses `H256`.  We have usage in two places one within zkEVM tracer (`H256`) and another while linking (`B256`) so one or the other would need to do conversion, but since we already define `bytecode_hash` as `H256` it's consistent on its parameters.